buildSupport/smallrye-openapi/src/main/kotlin/org/projectnessie/buildtools/smallryeopenapi/SmallryeOpenApiTask.kt

```diff
@@ -0,0 +1,449 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.projectnessie.buildtools.smallryeopenapi
+
+import io.smallrye.openapi.api.OpenApiConfig
+import io.smallrye.openapi.api.OpenApiDocument
+import io.smallrye.openapi.api.constants.OpenApiConstants
+import io.smallrye.openapi.runtime.OpenApiProcessor
+import io.smallrye.openapi.runtime.OpenApiStaticFile
+import io.smallrye.openapi.runtime.io.Format
+import io.smallrye.openapi.runtime.io.OpenApiSerializer
+import io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner
+import java.io.File
+import java.io.IOException
+import java.net.MalformedURLException
+import java.net.URL
+import java.net.URLClassLoader
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+import java.util.*
+import java.util.stream.Collectors
+import javax.inject.Inject
+import org.eclipse.microprofile.openapi.OASConfig
+import org.eclipse.microprofile.openapi.models.OpenAPI
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.NamedDomainObjectProvider
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.*
+import org.gradle.api.tasks.Optional
+import org.gradle.kotlin.dsl.listProperty
+import org.gradle.kotlin.dsl.property
+import org.jboss.jandex.IndexView
+
+@CacheableTask
+abstract class SmallryeOpenApiTask
```

Comment:
I assume there's no "native" gradle plugin for Smallrye?

Reply:
Sadly, yes